lib/hanami/cli/commands/app/generate/command.rb

```diff
@@ -38,9 +38,23 @@ def generator_class
               # fs:, inflector:, out:
             end
 
+            def detect_slice_from_pwd
+              current_dir = Pathname.new(Dir.pwd)
+              app_root = app.root
+              slices_dir = app_root.join("slices")
+              return nil unless current_dir.to_s.start_with?(slices_dir.to_s)
+
+              relative_path = current_dir.relative_path_from(slices_dir)
+              slice_name = relative_path.to_s.split("/").first
+              return slice_name if app.slices[slice_name.to_sym]
+
+              nil
```

Comment:
```suggestion
              slice_name if app.slices.keys.include?(slice_name.to_sym)
```
We can use implicit return instead of explicit (also returning nil when condition is false). 

Additionally, `app.slices[...]` raises a Hanami::SliceLoadError when it's not found, so we need to check on the keys. Can you add tests for this path? Likely by keeping the `app.slices[...]` and ensuring the error is raised, then modifying the code so it doesn't raise anymore.

Reply:
Done